src/Http/MiddlewareQueue.php

```diff
@@ -70,14 +78,22 @@ public function __construct(array $middleware = [])
     protected function resolve($middleware): MiddlewareInterface
     {
         if (is_string($middleware)) {
-            $className = App::className($middleware, 'Middleware', 'Middleware');
-            if ($className === null) {
-                throw new RuntimeException(sprintf(
-                    'Middleware "%s" was not found.',
-                    $middleware
-                ));
+            if ($this->container && $this->container->has($middleware)) {
+                $middleware = $this->container->get($middleware);
+            }
+
+            if (is_string($middleware)) {
```

Comment:
```suggestion
            } else {
```

Seems like this condition is redundant if we use `else`.